src/compiler/scala/tools/nsc/CompilerCommand.scala

```diff
@@ -111,7 +111,7 @@ class CompilerCommand(arguments: List[String], val settings: Settings) {
       val components = global.phaseNames // global.phaseDescriptors // one initializes
       s"Phase graph of ${components.size} components output to ${genPhaseGraph.value}*.dot."
     }
-    else allSettings.filter(_.isHelping).map(_.help).mkString("\n\n")
+    else allSettings.valuesIterator.filter(_.isHelping).map(_.help).mkString("\n\n")
```

Comment:
In the name of nominal programming, it's too bad that settings have values and all settings has a values iterator which are settings and not values. Such is life. Too bad it didn't flatmap over the help here.